java/feathub-udf/flink-udf/src/main/java/com/alibaba/feathub/flink/udf/SlidingWindowUtils.java

```diff
@@ -77,6 +81,12 @@ public static DataStream<Row> applySlidingWindowPreAggregationProcess(
             table = table.filter(Expressions.callSql(windowDescriptor.filterExpr));
         }
 
+        if (windowDescriptor.limit != null) {
+            aggDescriptors =
+                    getAggregationFieldsDescriptorsWithLimit(
```

Comment:
Currently for either Java or Python code, the concept of "limit" only exists on `WindowDescriptor`, not on `AggregationFieldDescriptor`. According to offline discussions with @Sxnan, a complete optimization of this problem might be as follows:

1. remove `limit` parameter from `WindowDescriptor`.
2. add `limit` parameter to `AggregationFieldDescriptor`, and get `aggFuncWithoutRetract` during instantiation as described in the comment.
3. handle aggregation fields with different `limit` values together in one sliding/over window evaluation process, instead of handling them separately and joining them afterward.

Given that it might be improper only to apply the second step above in this PR, and the complete steps seem to big to be put in this PR. I have created #97 to track this problem and maybe we can fix this in the next PR.